muxtools/subtitle/font.py

```diff
@@ -143,8 +192,115 @@ def clean_name(fontname: str) -> str:
+                    warn(f"No characters used in font '{fontname}'. Skipping subsetting...", collect_fonts)
+                
+                else:
+                    new_path = outpath.with_name(f"{outpath.stem}_subset{outpath.suffix}")
+
+                    old_font_family_name = ''
+                    new_font_family_name = ''
+                    old_font_name = ''
+                    new_font_name = ''
+
+                    try:
+                        font = ttLib.TTFont(outpath)
+
+                        subsetter = Subsetter()
+                        subsetter.populate(text="".join(characters_set))
+                        subsetter.subset(font)
+
+                        name_table = font["name"]
+
+                        for record in name_table.names:
+                            # Font names can only be 31 characters long
+
+                            if record.nameID == 1:  # Font Family name
+                                old_font_family_name = record.toUnicode().strip()
+                                new_font_family_name = record.toUnicode().replace(' ', '')[:25] + "Subset"
+                                record.string = new_font_family_name
+                            elif record.nameID == 4:  # Full name
+                                old_font_name = record.toUnicode().strip()
+                                new_font_name = record.toUnicode().replace(' ', '')[:25] + "Subset"
+                                record.string = new_font_name
+                            elif record.nameID == 6:  # PostScript name
+                                record.string = record.toUnicode().replace(' ', '')[:25] + "Subset"
+                        
+                        if not old_font_name or not new_font_name:
+                            raise Exception("Could not find necessary name records for subsetting.")
+                        
+                        font.save(new_path)
+                        font.close()
+                    
+                    except Exception as e:
+                        danger(f"Failed to subset font '{fontname}' (possibly corrupt/invalid font): {log_escape(str(e))}", collect_fonts)
+                    
+                    else:
+                        # It's important that the family name and font name are both replaced.
+                        # Usually, the family name is used in styles, but sometimes the full font name is used.
+                        
+                        # Insertion order is important as well, to avoid partial replacements.
+                        # For example:
+                        # - old_family_name = "Arial"
+                        # - old_font_name = "Arial Bold"
+                        # If we replaced "Arial" first, then "Arial Bold":
+                        # - \fnArial Bold -> \fnArialSubset Bold
+                        # which would not match the new name.
+                        # We need to replace the one which is longer first.
```

Comment:
this could be avoided if you guarantee only changing *full* tags, and (AFAICT) isn't even entirely reliable wrt. font selection ordering (see below)